QWeb/internal/frame.py

```diff
@@ -78,16 +78,21 @@ def wait_page_loaded() -> None:
                 raise QWebBrowserError(e) from e
             driver.switch_to.default_content()
     timeout = CONFIG["XHRTimeout"]
-    if timeout.lower() == "none":
-        return
+
     try:
         strategy = CONFIG["WaitStrategy"]
         if strategy.lower() == "legacy":
             logger.debug("Using legacy jQuery-based waiter")
+            if timeout.lower() == "none":
+                return
             xhr.wait_xhr_legacy(timestr_to_secs(timeout))
         else:
             logger.debug("Using enhanced waiter (network/spinner/render)")
-            xhr.wait_xhr(timestr_to_secs(timeout))
+            if timeout.lower() == "none":
```

Comment:
Is `timeout` guaranteed to be `str` or can it be type `None`?
Might be a good idea to cast it into a `str` first

Reply:
that's a good point and based on my understanding it's not prevented anywhere that user could do:
`SetConfig     XHRTimeout    ${None}`

...even if docs specifically say set it to (string) "None" to turn it "off".

Better add some guard / check there. Will fix with a new commit.

Reply:
fixed with new commit